recipes/protobuf/all/test_package/conanfile.py

```diff
@@ -6,12 +6,15 @@ class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
     generators = "cmake", "cmake_find_package_multi"
 
+    def build_requirements(self):
+        if tools.cross_building(self.settings):
+            self.build_requires(str(self.requires['protobuf']))
```

Comment:
May I ask, how does it work?

Reply:
Sure!
When cross building you need a `protoc` that is built for the build context, i.e "build_requires".
The `str(self.requires['protobuf'])` is there to automatically pick out the same version/channel
of protobuf from requires and mirror it to build_requires.